src/binding/petsc4py/src/petsc4py/PETSc/Mat.pyx

```diff
@@ -1799,23 +2906,98 @@ cdef class Mat(Object):
             asDims(dims, &cstarts[0], &cstarts[1], &cstarts[2])
         CHKERR( MatSetStencil(self.mat, ndim, cdims, cstarts, ndof) )
 
-    def setValueStencil(self, row, col, value, addv=None):
-        cdef _Mat_Stencil r = row, c = col
+    def setValueStencil(
+        self,
+        row: Stencil,
+        col: Stencil,
```

Comment:
I forgot to add explicit C types. Otherwise, if you pass `None` it will eventually segfault.

```suggestion
        MatStencil row: Stencil,
        MatStencil col: Stencil,
```